saleor/graphql/checkout/types.py

```diff
@@ -344,26 +359,66 @@ def resolve_lines(root: models.Checkout, info):
     # TODO: We should optimize it in/after PR#5819
     def resolve_available_shipping_methods(root: models.Checkout, info):
         def calculate_available_shipping_methods(data):
-            address, lines, discounts = data
-            available = get_valid_shipping_methods_for_checkout(root, lines, discounts)
+            address, lines, discounts, channel = data
+            channel_slug = channel.slug
+            display_gross = info.context.site.settings.display_gross_prices
+            manager = info.context.plugins
+            subtotal = manager.calculate_checkout_subtotal(
+                root, lines, address, discounts
+            )
+            if not address:
+                return []
+            available = get_valid_shipping_methods_for_checkout(
```

Comment:
This method can generate additional queries. Should we optimize it in this PR or separate PR?

Reply:
I would say a separate PR for that. Those shipping methods are used for both checkouts and orders and if you change one of them, you need to change many more places.

Reply:
You have right but this function in big and is used in many places. We should fix it in seperate PR.